modules/codegen/src/smithy4s/codegen/internals/Renderer.scala

```diff
@@ -892,11 +892,47 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
     )
   }
 
-  private def caseName(alt: Alt): NameRef = alt.member match {
-    case UnionMember.ProductCase(product) => NameRef(product.name)
-    case UnionMember.TypeCase(_) | UnionMember.UnitCase =>
-      NameRef(alt.name.dropWhile(_ == '_').capitalize + "Case")
-  }
+  private def caseName(unionName: NameRef, alt: Alt): NameRef =
```

Comment:
wondering if it might be beneficial to add a fold method on Union , as this pattern is repeated 4 times

Reply:
Idk I think the hit to overall readability is substantial with something like the following vs a pattern match:

```scala
private def caseNameForMatch(unionName: NameRef, alt: Alt): NameRef =
    alt.member.fold(_ =>
      unionName.down(alt.name.dropWhile(_ == '_').capitalize + "Case")
    )(product => unionName.down(product.name))(
      unionName.down(alt.name.dropWhile(_ == '_').capitalize + "Case.type")
    )
```